packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -1356,4 +1460,635 @@ it("Close vault LP and open new one-siced LP with just oSQTH ", async () => {
+      amount1Min: 0,
+      deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+    })
+    const wPowerPerpAmountInLPBefore = (isWethToken0) ? amount1 : amount0;
+    const wethAmountInLPBefore = (isWethToken0) ? amount0 : amount1;
+    // deposit nft to vault (approve first)
+    await shortSqueeth.connect(depositor).approve(controller.address, vaultId);
+    await controller.connect(depositor).updateOperator(vaultId, controllerHelper.address)
+    await (positionManager as INonfungiblePositionManager).connect(depositor).setApprovalForAll(controller.address, true) // approval for controller 
+    await (positionManager as INonfungiblePositionManager).connect(depositor).setApprovalForAll(controllerHelper.address, true) // approve controllerHelper
+    // Deposit nft to vault
+    await controller.connect(depositor).depositUniPositionToken(vaultId, tokenId)
+    const vaultBefore = await controller.vaults(vaultId);
+    // Double up LP
+    const wPowerPerpAmountToLp = wPowerPerpAmountInLPBefore
+    const wethAmountToLp = wethAmountInLPBefore
+    // Setup rebalanceVaultNft call
+    const abiCoder = new ethers.utils.AbiCoder
+    // Start with overcollateralized vault with NFT
+    // - Withdraw some eth (WithdrawFromVault)
+    // - Mint some squeeth (DepositIntoVault)
+    // - Increase LP liquidity (IncreaseLpLiquidity)
+
+    const rebalanceVaultNftParams = [
+      {
+        // Withdraw from vault
+        rebalanceVaultNftType: BigNumber.from(3), // WithdrawFromVault
+        // withdrawFromVault: [wPowerPerpToBurn, collateralToWithdraw, burnExactRemoved ]
+        data: abiCoder.encode(["uint256", 'uint256', 'bool'],
+         [BigNumber.from(0), wethAmountToLp, false ])
+        },
+      {
+        // Deposit into vault (no deposit)
+        rebalanceVaultNftType: BigNumber.from(2), // DepositIntoVault
+        // DepsositIntoVault: [wPowerPerpToMint, collateralToDeposit]
+        data: abiCoder.encode(["uint256", 'uint256'],
+         [wPowerPerpAmountToLp, BigNumber.from(0)])
+        },
+      {
+        // IncreaseLpLiquidity
+        rebalanceVaultNftType: BigNumber.from(0), // IncreaseLpLiquidity:
+        // IncreaseLpLiquidityParam: [tokenId, wPowerPerpAmountToLp, collateralToDeposit, wethAmountToLp, amount0Min, amount1Min]
+        data: abiCoder.encode(["uint256", 'uint256', 'uint256', 'uint256', 'uint256','uint256'],
+         [tokenId, wPowerPerpAmountToLp, BigNumber.from(0), wethAmountToLp, BigNumber.from(0), BigNumber.from(0)])
+        }
+    ]
+    // Flashloan to cover complete removal of LP (rearrange collateral ratio formula for 1.5 and add 0.1 ETH safety margin)
+    const normFactor = await controller.getExpectedNormalizationFactor()
+    const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+    const squeethPrice = await oracle.getTwap(wSqueethPool.address, wSqueeth.address, weth.address, 1, true)
+    const flashLoanAmount = vaultBefore.collateralAmount
```

Comment:
in this case, do we even need a flashloan, we had it collateralized at 300%, right?